enhancements/scheduling/scheduler-profiles.md

```diff
@@ -0,0 +1,208 @@
+---
+title: scheduling-profiles
+authors:
+  - "@damemi"
+reviewers:
+  - "@maszulik"
+  - "@ingvagabund"
+  - "@deads2k"
+approvers:
+  - "@maszulik"
+  - "@ingvagabund"
+  - "@deads2k"
+creation-date: 2020-11-11
+last-updated: 2020-11-11
+status: provisional
+see-also:
+  - "https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/624-scheduling-framework/kep.yaml"
+  - "/enhancements/kube-apiserver/audit-policy.md"
+replaces:
+superseded-by:
+---
+
+# Scheduling Profiles
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes a new field in the `schedulers.config.openshift.io/v1` API to allow
+configuration of enabled and disabled scheduler plugins via certain defined "profiles".
+
+This will allow users to configure pod scheduling beyond the default set of enabled plugins. It
+is effectively the equivalent to current scheduler Policy config, which has been deprecated.
+
+## Motivation
+
+As of Kubernetes 1.19, plugin profile configuration has replaced Policy files for the scheduler.
+We currently support setting scheduler predicates and priorities through the Policy API, and need
+to offer similar configuration with the new API in order to maintain a level of feature parity.
+
+However, the new plugin API is more expansive than the old API and provides more configuration possibilities
+than is necessary for most customers or reasonable to support. For those reasons we propose an
+abstracted representation of the most common plugin configurations for users to configure.
+
+### Goals
+
+1. define several common scheduler plugin configurations
+2. add a `profile` API field to `schedulers.config.openshift.io/v1` to specify which predetermined
+profile to use
+3. enable these profiles by kube-scheduler-operator
+
+### Non-Goals
+
+* configuration of individual scheduler plugins
+* platform support for custom schedulers or out-of-tree plugins
+
+## Proposal
+
+This enhancement proposes a new field in `schedulers.config.openshift.io/v1` named `profile`. This field
```

Comment:
Is there going to be a mapping of profiles to plugin configuration? Is the mapping going to be hardcoded in the operator? Will the operator allow to create a custom profile (e.g. through a configmap with customized configuration of plugins)? If not for customers, maybe for developers. Based on

> The available values will be constrained to those defined above by API validation, but there is the possibility of adding more options if significant use cases arise for different configurations.

and

> However, due to that there is a risk that this regresses from the current depth of configuration supported for customers with very specific needs (based on metrics gathered, there are likely very few clusters that would be impacted in this way)

it does not seem so.

Reply:
The mapping will be hardcoded (see https://github.com/openshift/enhancements/pull/541/files#r525406252) and will not allow any customization. Developers do have other methods for working on scheduler code, so that shouldn't be a problem too

Reply:
Yes, hardcoded again see https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/bindata/v4.1.0/kube-apiserver/audit-policies-cm.yaml how it's done for audit profiles.